Harvest.Web/ClientApp/src/Projects/ProjectDetailContainer.test.tsx

```diff
@@ -0,0 +1,136 @@
+    json: () => Promise.resolve(fakeProject),
+  };
+
+  const unbilledResponse = {
+    status: 200,
+    ok: true,
+    text: () => Promise.resolve("0.00"),
+  };
+
+  const fileResponse = {
+    status: 200,
+    ok: true,
+    text: () => Promise.resolve("file 1"),
+  };
+
+  const invoiceResponse = {
+    status: 200,
+    ok: true,
+    json: () => Promise.resolve(fakeInvoices),
+  };
+
+  const ticketResponses = {
+    status: 200,
+    ok: true,
+    json: () => Promise.resolve(fakeTickets),
+  };
+
+  it("Shows loading screen", async () => {
+    await act(async () => {
+      render(
+        <AppContext.Provider value={(global as any).Harvest}>
+          <MemoryRouter initialEntries={["/project/details/3"]}>
+            <Route path="/project/details/:projectId">
+              <ProjectDetailContainer />
+            </Route>
+          </MemoryRouter>
+        </AppContext.Provider>,
+        container
+      );
+    });
+
+    const messageContent = container.querySelector("div")?.textContent;
+    expect(messageContent).toContain("Loading");
+  });
+
+  it("Load details", async () => {
+    await act(async () => {
+      global.fetch = jest
+        .fn()
+        .mockImplementationOnce(() => Promise.resolve(projectResponse))
+        .mockImplementationOnce(() => Promise.resolve(unbilledResponse))
+        .mockImplementationOnce(() => Promise.resolve(ticketResponses))
+        .mockImplementationOnce(() => Promise.resolve(invoiceResponse))
+        .mockImplementationOnce(() => Promise.resolve(fileResponse));
+
+      render(
+        <AppContext.Provider value={(global as any).Harvest}>
+          <MemoryRouter initialEntries={["/project/details/3"]}>
+            <Route path="/project/details/:projectId">
+              <ProjectDetailContainer />
+            </Route>
+          </MemoryRouter>
+        </AppContext.Provider>,
+        container
+      );
+    });
+
+    const fieldTitle = container.querySelector("#request-title")?.textContent;
+    expect(fieldTitle).toContain("Field Request #3");
+  });
+
+  it("Display correct number of attachments", async () => {    
+    await act(async () => {
+      global.fetch = jest
+        .fn()
+        .mockImplementationOnce(() => Promise.resolve(projectResponse))
+        .mockImplementationOnce(() => Promise.resolve(unbilledResponse))
+        .mockImplementationOnce(() => Promise.resolve(ticketResponses))
+        .mockImplementationOnce(() => Promise.resolve(invoiceResponse))
+        .mockImplementationOnce(() => Promise.resolve(fileResponse));
+
+      render(
+        <AppContext.Provider value={(global as any).Harvest}>
+          <MemoryRouter initialEntries={["/project/details/3"]}>
+            <Route path="/project/details/:projectId">
+              <ProjectDetailContainer />
+            </Route>
+          </MemoryRouter>
+        </AppContext.Provider>,
+        container
+      );
+    });
+
+    const attachmentList = container.querySelector(".no-list-style");
```

Comment:
@wongband change this to attached-files-list to be more specific

Reply:
Sure thing!